IanSkelskey/evergit
# v0.1.3 Enhanced Buffer Handling for Large Diffs in evergit commit Command

### Summary
This pull request addresses the ENOBUFS error encountered when executing the evergit commit command with a large number of staged changes, which previously caused spawnSync to exceed its default buffer size.

### Changes Made
- Increased the buffer size in the getDiffForStagedFiles function to 10 MB, allowing it to handle larger diffs without failing.
- Added custom error handling for buffer overflow scenarios. If the output exceeds the buffer limit, a specific error message is thrown, guiding the user to limit staged files or increase the buffer size.

### Testing Instructions
- Prepare a Large Diff: Stage a significant number of changes or a large file.
- Run evergit commit: Execute the command to confirm that the increased buffer handles large diffs as expected.
- Custom Error Validation: Adjust the maxBuffer size to a lower value (e.g., 1 MB) and confirm that the custom error message appears if the diff output exceeds this limit.

### Additional Notes
This enhancement ensures evergit can handle larger commit diffs more reliably, and the custom error message improves the user experience by providing clear guidance when the buffer size is exceeded.
